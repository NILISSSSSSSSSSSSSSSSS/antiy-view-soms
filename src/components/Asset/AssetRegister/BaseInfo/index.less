.asset-base-info{
  background-color: #151B30;
  padding: 0 40px;
  min-height: 240px;
  .ant-form-item-label > label{
    color:#98ACD9;
  }
}

