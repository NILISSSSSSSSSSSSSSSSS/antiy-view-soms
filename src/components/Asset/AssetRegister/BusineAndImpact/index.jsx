import { Component } from 'react'
import { chunk } from 'lodash'
import { Row, Col, Form, Select } from 'antd'
import './style.less'

const Option  =  Select.Option

let key = 0
export default class BusineAndImpact extends Component {
  static defaultProps = {
    includesType: ['ip', 'mac']
  }
  constructor (props){
    super(props)
    this.state = { ipAndMacs: [{ ip: '', mac: '', key: key }] }
  }
  /**
   * 根据表单项的类型，渲染表单项
   * @param filed 表单项
   * @param rowIndex 表单项索引
   * @param colIndex 表单项索引
   * @param ipAndMac{Object} 一行的ip和mac对象数据
   */
  renderFormItem = (filed, rowIndex, colIndex, ipAndMac) => {
    const { form, FormItem, formLayout = {} } = this.props
    const { ipAndMacs } = this.state
    const { getFieldDecorator } = form || {}
    const { type, name, key, rules, placeholder = '请输入', data, config, defaultValue: initialValue, popover, ...other } = filed || {}
    const { allowClear, disabled } = other
    // 默认允许删除
    const _allowClear = { allowClear: true }
    //设置不允许被删除或者禁用时，不允许被删除
    if(typeof allowClear === 'boolean' && !allowClear || disabled) {
      _allowClear.allowClear = false
    }
    if(type === 'operation'){
      return (
        <div className="bus_imp_operation">
          { ipAndMacs.length > 1 ? <img src={require('@/assets/biaodanshanchu.svg')} onClick={()=>{this.del(rowIndex)}} alt="" /> : null }
          <img src={require('@/assets/biaodanxinzeng.svg')} onClick={()=>{this.add(rowIndex)}} alt="" />
        </div>
      )
    }

    return (
      <FormItem label={ name } { ...formLayout }>
        { getFieldDecorator(`iaAndMac${ipAndMac.key}${colIndex}`, {
          rules,
          initialValue
        })(
          <Select
            showSearch
            optionFilterProp="children"
            getPopupContainer={triggerNode => triggerNode.parentNode}
            placeholder="请选择" >
            {
              data.length && data.map((item, index) => {
                return (<Option key={item.id}>{item.value}</Option>)
              })
            }
          </Select>
        ) }
      </FormItem>
    )
  }
  /**
   * ip和mac变更
   * @param rowIndex
   * @param colIndex
   * @param value
   */
  onChnage = (rowIndex, colIndex, value) => {
    const { onChange } = this.props
    const { ipAndMacs } = this.state
    const _ipAndMacs = ipAndMacs.map((e, i)=>{
      if(i === rowIndex){
        return { ...e, [!colIndex ? 'ip' : 'mac']: value }
      }
      return e
    })
    this.setState({ ipAndMacs: _ipAndMacs })
    onChange && onChange(_ipAndMacs)
  }
  /**
   * 删除ip和mac
   * @param rowIndex
   */
  del = (rowIndex) => {
    const { onChange } = this.props
    const { ipAndMacs } = this.state
    const _ipAndMacs = [].concat(ipAndMacs)
    _ipAndMacs.splice(rowIndex, 1, )
    onChange && onChange(_ipAndMacs)
    this.setState({ ipAndMacs: _ipAndMacs })
  }
  /**
   * 添加ipmac
   * @param rowIndex
   */
  add = (rowIndex) => {
    const { onChange } = this.props
    const { ipAndMacs } = this.state
    const _ipAndMacs = [].concat(ipAndMacs)
    _ipAndMacs.splice(rowIndex + 1, 0, { key: ++key })
    onChange && onChange(_ipAndMacs)
    this.setState({ ipAndMacs: _ipAndMacs })
  }
  renderDom = (ipAndMacs, fields2, column) => {
    return ipAndMacs.map((ipAndMac, i)=>{
      return (
        <Form key={ipAndMac.key}>
          <Row>
            <div className="form-wrap">
              {
                chunk(fields2, column).map((el) => {
                  return (
                    el.map((it, idx) => {
                      return (
                        <Col key={ `${it.key}${i}` } span={ 24 / column }>
                          { this.renderFormItem(it, i, idx, ipAndMac) }
                        </Col>
                      )
                    })
                  )
                })
              }
            </div>
          </Row>
        </Form>
      )
    })
  }
  render () {
    const { column } = this.props
    const { ipAndMacs } = this.state
    const fields2 = [
      { name: '从属业务', key: 'ip2', type: 'select', data: [{ id: 1, value: '支付' }, { id: 2, value: '语音' }, { id: 3, value: '视屏' }] },
      { name: '业务影响', key: 'mac2', type: 'select', data: [{ id: 1, value: '高' }, { id: 2, value: '中' }, { id: 3, value: '低' }]  },
      { name: '', key: '', type: 'operation' }
    ]
    return (
      <div className="form-content">
        { this.renderDom(ipAndMacs, fields2, column)}
      </div>
    )
  }
}
