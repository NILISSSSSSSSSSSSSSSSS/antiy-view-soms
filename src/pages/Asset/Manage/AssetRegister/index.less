@import '../../../../assets/less/vars.less';
.asset-register-container{
  position: relative;
  //padding-top: 40px !important;
  .asset-register-template-header{
    font-size: 14px;
    color: @detail-label-color;
  }
  .font-step-info{
    // background: #FFDEB5;
    color: #C28335;
    letter-spacing: 0;
    line-height: 20px;
    //height: 40px;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: flex-start;
    //position: absolute;
    left: 0;
    top: 0;
    padding: 10px 40px;
    width: 100%;
    .font-step-info-note-box, .font-step-info-adjunct-box{
      width: 100%;
      position: relative;
      display: flex;
    }
    .font-step-info-note-box .font-step-info-note, .font-step-info-adjunct-box .font-step-info-adjunct{
      width: calc(~'100% - 74px');
      white-space: nowrap;
      text-overflow: ellipsis;
      overflow: hidden;
      display: inline-block;
      //position: absolute;
      left: 74px;
      top: 0;
      .ant-tooltip-inner{
        white-space: normal;
      }
    }
    .font-step-info-adjunct{
      display: flex !important;
      flex-direction: column;
    }
    .font-step-info-adjunct>div{
      text-overflow: ellipsis;
      overflow: hidden;
      display: inline-block;
    }
    .font-step-info-label{
      width: 74px;
    }
    .picture{
      color: #3469FF;
      cursor: pointer;
    }
  }
  .custom-panel-header{
    display: flex;
    justify-content: space-between;
    align-items: center;
    border-bottom: 1px solid #506099;
    padding-bottom: 11px;
    color: #FFFFFF;
    .custom-panel-header-unfold{
      color: #3B6CFF;
      cursor: pointer;
    }
  }

  // 重新定义手风琴样式
  .ant-collapse.custom-collapse{
    padding: 20px 40px;
    background: #151B30;
    .ant-collapse-header{
      padding-left: 0;
      i{
        display: none;
      }
    }
    .ant-collapse-item{
      border: 0;
    }
  }

}
