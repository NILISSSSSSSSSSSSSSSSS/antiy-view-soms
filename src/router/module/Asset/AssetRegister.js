import Loadable from 'react-loadable'
import Loading from '@/components/common/Loading'

const AssetRegister = Loadable({
  loader: () => import('@/pages/Asset/Manage/AssetRegister'),
  loading: Loading
})

export default AssetRegister
