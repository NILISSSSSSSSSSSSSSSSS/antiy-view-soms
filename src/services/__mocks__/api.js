// GET 根据ID获取资产详情

export function getAssetHardWareById (params) {
  return Promise.resolve({
    head: { code: '200' },
    body: { asset: { stringId: '-------------' } }
  })
}

export function batchRelation (params) {
  return Promise.resolve({
    head: { code: '200' },
    body: { }
  })
}

export function claimTaskBatch (params) {
  return Promise.resolve({
    head: { code: '200' }
  })
}

export function querySuddenList (params) {
  return Promise.resolve({
    head: { code: '200' },
    body: {
      items: [
        {
          antiyVulnId: '12345'
        }
      ],
      totalRecords: 1
    }
  })
}
export function getList (params) {
  return Promise.resolve({
    head: { code: '200' },
    body: {
      items: [{}],
      totalRecords: 1
    }
  })
}
export function scanUrl (params) {
  return Promise.resolve({
    head: { code: '200' },
    body: {
      onProgress: 10,
      errorMessage: '',
      findAssetNums: 10
    }
  })
}
export function getBugUsers (params) {
  return Promise.resolve({
    head: { code: '200' },
    body: []
  })
}
export function onBackSubmit (params) {
  return Promise.resolve({
    head: { code: '200' },
    body: []
  })
}
export function getAdvise (params) {
  return Promise.resolve({
    head: { code: '200' },
    body: {}
  })
}
export function getVulnCpe (params) {
  return Promise.resolve({
    head: { code: '200' },
    body: {
      items: [],
      totalRecords: 1
    }
  })
}
export default {
  scanUrl,
  getAdvise,
  onBackSubmit,
  getBugUsers,
  getList,
  getAssetHardWareById,
  batchRelation,
  claimTaskBatch,
  querySuddenList,
  getVulnCpe
}
