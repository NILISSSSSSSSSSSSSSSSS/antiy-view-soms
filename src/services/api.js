import asset from './assets'
import * as workOrderApi from './workOrder'
import * as systemApi from './system'
import * as safeApi from './safe'
import * as organizationApi from './organization'
import * as taskPlan from './taskPlan'
import * as taskPlanReport from './taskPlanReport'
import * as bug from './bug'
import log from  './log'
import * as knowledge from  './knowledge'
import * as inspection from  './inspection'
import alarm from './alarm'
import * as baseSetting from  './baseSetting'
import * as indexPage from  './indexPage'
import * as workBench from  './workBench'
import * as reportForm from  './reportForm'
import  personalCenter from  './personalCenter'
import patch from './patch'
import * as defend from  './defend'
import * as examine from  './examine'
export default {
  ...patch,
  ...workBench,
  ...indexPage,
  ...asset,
  ...workOrderApi,
  ...systemApi,
  ...taskPlan,
  ...taskPlanReport,
  ...organizationApi,
  ...bug,
  ...log,
  ...knowledge,
  ...alarm,
  ...baseSetting,
  ...safeApi,
  ...inspection,
  ...reportForm,
  ...personalCenter,
  ...defend,
  ...examine
}
