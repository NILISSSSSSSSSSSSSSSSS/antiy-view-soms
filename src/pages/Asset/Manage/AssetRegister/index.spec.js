import React from 'react'
import { shallow } from 'enzyme'
import AssetRegister from './index'
import api from '@/services/api'
import {
  COMPUTING_DEVICE,
  STORAGE_DEVICE,
  NETWORK_DEVICE,
  SAFETY_DEVICE,
  OTHER_DEVICE
} from '@a/js/enume'
// jest.mock('@/services/api')
const head = { code: '200' }
const body = {
  asset: {},
  assetNetworkEquipment: {},
  assetSafetyEquipment: {},
  assetStorageMedium: {}
}
api.getAssetHardWareById = jest.fn()
api.getAssetHardWareById.mockReturnValue(Promise.resolve({
  head,
  body
}))
describe('资产新登记页面 <AssetRegister />', () => {
  let wrapper
  const props = {
    location: {
      search: '?source=register&register=new'
    },
    form: {
      validateFields: (func) => {
        func()
      }
    },
    history: {
      push: jest.fn()
    }
  }
  beforeEach(() => {
    wrapper = shallow(<AssetRegister { ...props } />).dive()
  })
  // it('快照', ()=>{
  //   const app = wrapper.debug()
  //   expect(app).toMatchSnapshot()
  // })
  it('AssetRegister页面的正常渲染', () => {
    expect(wrapper.find('.asset-register-container')).toHaveLength(1)
  })
  it('baseChange  ===> 资产基础信息设备类型变化时', () => {
    const instance = wrapper.instance()
    const spy_baseChange = jest.spyOn(instance, 'baseChange')
    /**
     * 登记资产为计算设备时
     */
    // 原始数据
    wrapper.setState({ baseInfo: { operationSystem: '1', categoryModel: COMPUTING_DEVICE.value } })
    spy_baseChange({ operationSystem: '1', categoryModel: COMPUTING_DEVICE.value })
    // 需走配置流程
    expect(instance.isShowNextStep.categoryModel).toBeTruthy()
    // 重置state
    // wrapper.restore()
    /**
     * 登记资产时，一但切换资产类型，则认为模板、软件、组件都为选择，
     * 并且情况相应数据
     */
    // 原始数据
    wrapper.setState({ baseInfo: { operationSystem: '1', categoryModel: COMPUTING_DEVICE.value } })
    spy_baseChange({ operationSystem: '1', categoryModel: NETWORK_DEVICE.value })
    // 判断为非计算设备
    expect(instance.isShowNextStep.categoryModel).toBeFalsy()
    expect(instance.isShowNextStep.baseTemplate).toBeFalsy()
    expect(instance.isShowNextStep.software).toBeFalsy()
    expect(instance.isShowNextStep.disk).toBeFalsy()
    expect(instance.templateVerificationFuc).toBeNull()
    expect(instance.subassemblyList).toEqual([])
    expect(wrapper.state().baseInfo.categoryModel).toEqual(NETWORK_DEVICE.value)
    expect(wrapper.state().baseInfo.templates).toEqual([])
    // 重置state
    // wrapper.restore()
    /**
     * 切换资产版本时
     */
    const spy_getCPEAssemblyList = jest.spyOn(instance, 'getCPEAssemblyList')
    // 原始数据
    wrapper.setState({ baseInfo: { businessId: '1' } })
    spy_baseChange({ businessId: '2' })
    expect(spy_getCPEAssemblyList).toHaveBeenCalled()
    spy_baseChange({ businessId: null })
    expect(wrapper.state().defaultValueSubassemblyList).toEqual([])
  })
  it('templateChange ===》模板改变事件', () => {
    const instance = wrapper.instance()
    const spy_templateChange = jest.spyOn(instance, 'templateChange')
    /**
     * 选择模板时，这会出现下一步步骤
     */
    let temps = [{}]
    spy_templateChange(temps)
    expect(instance.isShowNextStep.baseTemplate).toBeTruthy()
    /**
     * 没有选择模板时，则不显示下一步骤
     */
    temps = []
    spy_templateChange(temps)
    expect(instance.isShowNextStep.baseTemplate).toBeFalsy()
  })
  it('renderFrontStep ===》渲染上一步的驳回信息', () => {
    const instance = wrapper.instance()
    const spy_renderFrontStep = jest.spyOn(instance, 'renderFrontStep')
    /**
     * 没有驳回信息时
     */
    wrapper.setState({ preNodeInfo: {} })
    let dom = spy_renderFrontStep()
    expect(dom).toBeNull()
    /**
     * 有驳回信息时
     */
    const fileInfo = JSON.stringify([{ fileName: '测试' }])
    wrapper.setState({ preNodeInfo: { note: '通过', fileInfo, originStatus: 8 } })
    dom = spy_renderFrontStep()
    expect(dom).toMatchSnapshot()
  })
  it('generatorDisabled ===》模板是否被禁用', () => {
    const instance = wrapper.instance()
    const spy_renderFrontStep = jest.spyOn(instance, 'generatorDisabled')
    // 安全设备
    let flag = spy_renderFrontStep(SAFETY_DEVICE.value)
    expect(flag).toBeFalsy()
    // 计算设备
    flag = spy_renderFrontStep(COMPUTING_DEVICE.value, '1')
    expect(flag).toBeFalsy()
    // 网络设备
    flag = spy_renderFrontStep(NETWORK_DEVICE.value)
    expect(flag).toBeFalsy()
    // 存储设备
    flag = spy_renderFrontStep(STORAGE_DEVICE.value)
    expect(flag).toBeFalsy()
    // 其他设备
    flag = spy_renderFrontStep(OTHER_DEVICE.value)
    expect(flag).toBeFalsy()
  })
  it('collapseOnChange ===》展开collapse事件', () => {
    const instance = wrapper.instance()
    const spy_collapseOnChange = jest.spyOn(instance, 'collapseOnChange')
    // 默认展开基础信息、模板信息
    expect(wrapper.state().activeKey).toEqual(['1', '2'])
    // 收起模板
    spy_collapseOnChange('2')
    expect(wrapper.state().activeKey.includes('2')).toBeFalsy()
    //展开组件信息
    spy_collapseOnChange('3')
    expect(wrapper.state().activeKey.includes('3')).toBeTruthy()
  })
  it('subassemblyChange  ===> 新登记时资产组件变化时', () => {
    const instance = wrapper.instance()
    const spy_subassemblyChange = jest.spyOn(instance, 'subassemblyChange')
    const DISK1 = { type: 'DISK', businessId: '1', amount: 1 }
    let addList = [DISK1]
    let list = [DISK1]
    /**
     * 新登记时任何资产类型，改变了任何组件都不会显示下一步骤
     */
    wrapper.setState({ baseInfo: { categoryModel: 1 } })
    // 添加了硬盘
    spy_subassemblyChange(list, addList, [], [], [])
    // 显示下一步操作
    expect(instance.isShowNextStep.disk).toBeFalsy()
  })
  it('getCPEAssemblyList  ===> 资产切换版本时', () => {
    const instance = wrapper.instance()
    api.getCPEAssemblyList = jest.fn()
    api.getCPEAssemblyList.mockReturnValue(Promise.resolve({ head, body: [] }))
    const spy_getCPEAssemblyList = jest.spyOn(instance, 'getCPEAssemblyList')
    spy_getCPEAssemblyList()
  })
  it('onSubmit  ===> 点击提交按钮事件', () => {
    const nextStep = {
      nextStep: 'nextStep',
      nextExecutor: []
    }
    const instance = wrapper.instance()
    const spy_onSubmit = jest.spyOn(instance, 'onSubmit')
    const spy_templateChange = jest.spyOn(instance, 'templateChange')
    // 表单验证未通过
    props.form.validateFields = (func)=>{
      func(true, { })
    }
    spy_onSubmit(nextStep)
    // 表单验证通过
    props.form.validateFields = (func)=>{
      func(false, { })
    }
    /**
     * 提交计算设备，没有选择模板时,展开模板的选项
     */
    wrapper.setState({ baseInfo: { categoryModel: COMPUTING_DEVICE.value } })
    spy_onSubmit(nextStep)
    expect(wrapper.state().activeKey.includes('2')).toBeTruthy()
    /**
     * 模拟选择了模板
     */
    spy_templateChange([{}], ()=>false)
    spy_onSubmit(nextStep)

    /**
     * 依次验证其他资产类型的设备
     */
    wrapper.setState({ baseInfo: { categoryModel: 2 } })
    spy_onSubmit(nextStep)
    wrapper.setState({ baseInfo: { categoryModel: 3 } })
    spy_onSubmit(nextStep)
    wrapper.setState({ baseInfo: { categoryModel: 4 } })
    spy_onSubmit(nextStep)
    wrapper.setState({ baseInfo: { categoryModel: 5 } })
    spy_onSubmit(nextStep)
  })
  it('assetRegister  ===> 新登记请求提交', () => {
    const instance = wrapper.instance()
    const spy_assetRegister = jest.spyOn(instance, 'assetRegister')
    api.saveAsset = jest.fn()
    api.saveAsset.mockReturnValue(Promise.resolve({ head, body: '后台提示' }))
    spy_assetRegister({ asset: { nextStep: 'safetyCheck' } })

    api.saveAsset.mockReturnValue(Promise.resolve({ head }))
    spy_assetRegister({ asset: { nextStep: 'safetyCheck' } })
  })
})
describe('资产再次登记页面 <AssetRegister />', () => {
  const props = {
    location: {
      search: '?source=register&id=4bDspg871efwNHvK6%2FmuOg%3D%3D&register=again'
    },
    form: {
      validateFields: (func) => {
        func()
      }
    },
    history: {
      push: jest.fn()
    }
  }
  let wrapper
  beforeEach(() => {
    wrapper = shallow(<AssetRegister { ...props } />).dive()
  })
  // it('快照', ()=>{
  //   const app = wrapper.debug()
  //   expect(app).toMatchSnapshot()
  // })
  it('AssetRegister页面的正常渲染', () => {
    expect(wrapper.find('.asset-register-container')).toHaveLength(1)
  })
  it('subassemblyChange  ===> 资产组件变化时', () => {
    const instance = wrapper.instance()
    const spy_subassemblyChange = jest.spyOn(instance, 'subassemblyChange')
    const DISK1 = { type: 'DISK', businessId: '1', amount: 1 }
    const DISK2 = { type: 'DISK', businessId: '1', amount: 2 }
    const CPU = { type: 'CPU', amount: 2 }
    let addList = [DISK1]
    let list = [DISK1]
    /**
     * 资产类型为计算设备时并且包含硬盘时
     */
    wrapper.setState({ baseInfo: { categoryModel: 1 } })
    // 添加了硬盘
    spy_subassemblyChange(list, addList, [], [], [])
    // 显示下一步操作
    expect(instance.isShowNextStep.disk).toBeTruthy()
    /**
     * 资产类型为计算设备时并且包含硬盘时，数量不一致时
     */
    const changeList = [DISK2]
    const originalList = [DISK1]
    wrapper.setState({ baseInfo: { categoryModel: 1 } })
    // 添加了硬盘
    spy_subassemblyChange(changeList, [], [], changeList, originalList)
    // 显示下一步操作
    expect(instance.isShowNextStep.disk).toBeTruthy()
    /**
     * 资产类型为计算设备时不包含硬盘时
     */
    addList = [CPU]
    list = [CPU]
    wrapper.setState({ baseInfo: { categoryModel: 1 } })
    // 添加了硬盘
    spy_subassemblyChange(list, addList, [], [], [])
    // 显示下一步操作
    expect(instance.isShowNextStep.disk).toBeFalsy()
    /**
     * 资产类型为非计算设备时包含硬盘时
     */
    addList = [DISK1]
    list = [DISK1]
    wrapper.setState({ baseInfo: { categoryModel: 2 } })
    // 添加了硬盘
    spy_subassemblyChange(list, addList)
    // 不显示下一步操作
    expect(instance.isShowNextStep.disk).toBeFalsy()
    /**
     * 资产类型为非计算设备时不包含硬盘时
     */
    addList = [CPU]
    list = [CPU]
    wrapper.setState({ baseInfo: { categoryModel: 2 } })
    // 添加了硬盘
    spy_subassemblyChange(list, addList)
    // 不显示下一步操作
    expect(instance.isShowNextStep.disk).toBeFalsy()
  })
  it('assetRegister  ===> 再次登记请求提交', () => {
    const instance = wrapper.instance()
    const spy_assetRegister = jest.spyOn(instance, 'assetRegister')
    // 默认是自己的代办任务
    wrapper.setState({ assetData: { asset: { waitingTaskReponse: {} } } })
    const params = { asset: { nextStep: 'safetyCheck' }, manualStartActivityRequest: {} }
    api.changAsset = jest.fn()
    api.changAsset.mockReturnValue(Promise.resolve({ head, body: '后台提示' }))
    spy_assetRegister(params)

    api.changAsset.mockReturnValue(Promise.resolve({ head }))
    spy_assetRegister(params)
  })
})
describe('未知资产再次登记页面 <AssetRegister />', () => {
  const props = {
    location: {
      search: '?source=appear&id=4bDspg871efwNHvK6%2FmuOg%3D%3D&register=new'
    },
    form: {
      validateFields: (func) => {
        func()
      }
    },
    history: {
      push: jest.fn()
    }
  }
  let wrapper
  beforeEach(() => {
    wrapper = shallow(<AssetRegister { ...props } />).dive()
  })
  // it('快照', ()=>{
  //   const app = wrapper.debug()
  //   expect(app).toMatchSnapshot()
  // })
  it('AssetRegister页面的正常渲染', () => {
    expect(wrapper.find('.asset-register-container')).toHaveLength(1)
  })
})
