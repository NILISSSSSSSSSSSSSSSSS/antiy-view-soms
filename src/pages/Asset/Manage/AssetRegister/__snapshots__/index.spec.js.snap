// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`资产新登记页面 <AssetRegister /> renderFrontStep ===》渲染上一步的驳回信息 1`] = `
<div
  className="font-step-info"
>
  <div>
    整改备注
    <span
      className="font-step-info-note"
    >
      通过
    </span>
  </div>
  <div>
    附件：
    <span
      className="picture"
      onClick={[Function]}
    >
      测试
    </span>
  </div>
</div>
`;
